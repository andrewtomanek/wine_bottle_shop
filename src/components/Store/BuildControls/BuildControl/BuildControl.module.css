.BuildControl {
  display: grid;
  grid-template-columns: 3fr 1fr 1fr 1fr;
  grid-auto-flow: column;
  align-items: center;
  margin: 0;
  background-color: var(--red);
  border: 0.2rem solid var(--wine);
}

.BuildControl button {
  font: inherit;
  font-size: 1rem;
  font-weight: 900;
  padding: 0.3rem 0.6rem;
  margin: 0;
  border: 1px solid #aa6817;
  cursor: pointer;
  outline: none;
}

.BuildControl button:disabled {
  background-color: #ac9980;
  border: 1px solid #7e7365;
  color: #ccc;
  cursor: default;
}

.BuildControl button:hover:disabled {
  background-color: #ac9980;
  color: #ccc;
  cursor: not-allowed;
}

.Label {
  color: white;
  padding: 0.3rem;
  font-weight: 800;
  text-align: left;
}

.Price {
  background-color: var(--orange);
  color: white;
  padding: 0.3rem;
  font-weight: 600;
  text-align: right;
}

.BuildControl .Less {
  background-color: red;
  color: white;
}

.BuildControl .More {
  background-color: var(--green);
  color: white;
}

.BuildControl .Less:hover,
.BuildControl .Less:active {
  background-color: #daa972;
  color: white;
}

.BuildControl .More:hover,
.BuildControl .More:active {
  background-color: #99703f;
  color: white;
}

@media (min-width: 500px) {
  .BuildControl {
    display: grid;
    grid-template-columns: 5fr 1fr 1fr 2fr;
    grid-auto-flow: column;
    align-items: center;
    margin: 0;
    background-color: var(--red);
    border: 0.2rem solid var(--wine);
  }
}
