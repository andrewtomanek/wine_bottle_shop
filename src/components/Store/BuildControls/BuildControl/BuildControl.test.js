import React from "react";
import { shallow } from "enzyme";
import BuildControl from "./BuildControl";

const added = jest.fn();
const removed = jest.fn();

const props = {
  label: "Label-test",
  disabled: false,
  added,
  removed,
  price: 99,
};

describe("Shallow Render BuildControl component", () => {
  it("renders Label", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper.find(".Label").text()).toContain("Label-test");
  });

  it("renders - button", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper.find(".Less").text()).toContain("-");
  });

  it("renders + button", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper.find(".More").text()).toContain("+");
  });
  it("renders price", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper.find(".Price").text()).toContain("99");
  });

  it("renders 2 buttons", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper.find("button")).toHaveLength(2);
  });

  it("Match Snapshot of BuildControl component", () => {
    const wrapper = shallow(<BuildControl {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
