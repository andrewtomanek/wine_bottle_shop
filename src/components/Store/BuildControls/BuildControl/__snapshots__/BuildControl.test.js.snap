// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render BuildControl component Match Snapshot of BuildControl component 1`] = `
<div
  className="BuildControl"
>
  <div
    className="Label"
  >
    Label-test
  </div>
  <button
    className="Less"
    disabled={false}
    onClick={[MockFunction]}
  >
    -
  </button>
  <button
    className="More"
    onClick={[MockFunction]}
  >
    +
  </button>
  <div
    className="Price"
  >
    99
  </div>
</div>
`;
